src/blockdev.rs

```diff
@@ -461,6 +464,205 @@ impl Mount {
+        let disk = disk.as_ref();
+        let mut result = Self {
+            sector_size: None,
+            partitions: Vec::new(),
+        };
+
+        // skip reading partition table if we don't have an accept filter
+        if filters.is_empty() {
+            return Ok(result);
+        }
+
+        let mut f = OpenOptions::new()
+            .read(true)
+            .open(disk)
+            .chain_err(|| format!("opening {} for reading", disk.display()))?;
+        let gpt = match GPT::find_from(&mut f) {
+            Ok(gpt) => gpt,
+            Err(gptman::Error::InvalidSignature) => return Ok(result),
+            Err(e) => {
+                return Err(e)
+                    .chain_err(|| format!("reading partition table of {}", disk.display()))?
+            }
+        };
+        // true except in unit tests
+        if f.metadata()
+            .chain_err(|| format!("getting metadata for {}", disk.display()))?
+            .file_type()
+            .is_block_device()
+        {
+            let disk_sector_size = get_sector_size(&f)?.get() as u64;
+            if disk_sector_size != gpt.sector_size {
+                // give up before doing any damage
+                bail!(
+                    "sector size {} of disk {} doesn't match sector size {} of GPT",
+                    disk_sector_size,
+                    disk.display(),
+                    gpt.sector_size
+                );
+            }
+        }
+        result.sector_size = Some(gpt.sector_size);
+
+        for (i, p) in gpt.iter() {
+            if Self::accept(i, p, filters) {
+                result.partitions.push((i, p.clone()));
+            }
+        }
+        Ok(result)
+    }
+
+    fn accept(i: u32, p: &GPTPartitionEntry, filters: &[PartitionFilter]) -> bool {
+        use PartitionFilter::*;
+        if !p.is_used() {
+            return false;
+        }
+        filters.iter().any(|f| match f {
+            Index(Some(first), _) if first.get() > i => false,
+            Index(_, Some(last)) if last.get() < i => false,
+            Index(_, _) => true,
+            Label(glob) if glob.matches(p.partition_name.as_str()) => true,
+            _ => false,
+        })
+    }
+
+    pub fn write<P: AsRef<Path>>(&self, disk: P) -> Result<()> {
+        if self.partitions.is_empty() {
+            return Ok(());
+        }
+
+        // get or create GPT
+        let disk = disk.as_ref();
+        let mut f = OpenOptions::new()
+            .read(true)
+            .open(disk)
+            .chain_err(|| format!("opening {} for reading", disk.display()))?;
+        // true except in unit tests
+        if f.metadata()
+            .chain_err(|| format!("getting metadata for {}", disk.display()))?
+            .file_type()
+            .is_block_device()
+        {
+            let disk_sector_size = get_sector_size(&f)?.get() as u64;
+            if disk_sector_size != self.sector_size.expect("sector size unset") {
```

Comment:
Nit: I'd be happier if we could avoid this `expect` (and the matching `unwrap` below).